src/Tasks.UnitTests/Copy_Tests.cs

```diff
@@ -84,6 +84,38 @@ public void Dispose()
             Copy.RefreshInternalEnvironmentValues();
         }
 
+        [Fact]
+        [SkipOnPlatform(TestPlatforms.Windows, "The error message derived directly from the hardlink creation failure is not provided by MSBuild, so including what it actually logs would be artificial.")]
+        public void TestErrorOnHardlinkFailUnix()
```

Comment:
Can you put this in `CopyHardLink_Tests` instead of the base class so it isn't run multiple times?